graphql-server/src/schemas/queries/patients.ts

```diff
@@ -177,3 +190,88 @@ export async function queryDashboardPatients({
     await session.close();
   }
 }
+
+export async function queryPatientIdsTriplets(searchVals: Array<string>) {
+  const searchValsWithoutCDash = searchVals.map((searchVal) =>
+    searchVal.startsWith("C-") ? searchVal.slice(2) : searchVal
+  );
+  const searchValList = searchValsWithoutCDash
+    .map((searchVal) => `'${searchVal}'`)
+    .join(",");
+  const query = `
+    SELECT
+      CMO_PATIENT_ID,
+      DMP_PATIENT_ID,
+      MRN
+    FROM
+      ${PHI_ID_MAPPING_TABLE}
+    WHERE
+      DMP_PATIENT_ID IN (${searchValList})
+      OR MRN IN (${searchValList})
+      OR CMO_PATIENT_ID IN (${searchValList})
+  `;
+  const patientIdsTriplets = await queryDatabricks<PatientIdsTriplet>(query);
+  patientIdsTriplets.forEach((patientIdTriplet) => {
+    patientIdTriplet.CMO_PATIENT_ID = `C-${patientIdTriplet.CMO_PATIENT_ID}`;
+  });
+  return patientIdsTriplets;
+}
+
+export async function queryAnchorSeqDatesByDmpPatientId(
+  dmpPatientIds: Array<string>
+) {
+  const dmpPatientIdsList = dmpPatientIds
+    .map((dmpPatientId) => `'${dmpPatientId}'`)
+    .join(",");
+  const query = `
+    SELECT
+      DMP_PATIENT_ID,
+      ANCHOR_SEQUENCING_DATE
+    FROM
+      ${SEQ_DATES_BY_PATIENT_TABLE}
+    WHERE
+      DMP_PATIENT_ID IN (${dmpPatientIdsList})
+  `;
+  return await queryDatabricks<AnchorSeqDateByDmpPatientId>(query);
+}
+
+export function mapPhiToPatientsData({
+  patientsData,
+  patientIdsTriplets,
+  anchorSeqDatesByDmpPatientId,
+}: {
+  patientsData: DashboardPatient[];
+  patientIdsTriplets: Array<PatientIdsTriplet>;
+  anchorSeqDatesByDmpPatientId: Array<AnchorSeqDateByDmpPatientId>;
+}): Array<DashboardPatient> {
+  const mrnByCmoPatientIdMap: Record<string, string> = {};
+  const mrnByDmpPatientIdMap: Record<string, string> = {};
+  patientIdsTriplets.forEach((triplet) => {
+    if (triplet.MRN) {
+      mrnByCmoPatientIdMap[triplet.CMO_PATIENT_ID] = triplet.MRN;
+      if (triplet.DMP_PATIENT_ID) {
+        mrnByDmpPatientIdMap[triplet.DMP_PATIENT_ID] = triplet.MRN;
+      }
+    }
+  });
+  const anchorSeqDateByDmpPatientIdMap: Record<string, string> = {};
+  anchorSeqDatesByDmpPatientId.forEach((record) => {
+    anchorSeqDateByDmpPatientIdMap[record.DMP_PATIENT_ID] =
+      record.ANCHOR_SEQUENCING_DATE;
+  });
+  return patientsData.map((patient) => {
+    const cmoPatientId = patient.cmoPatientId;
+    const dmpPatientId = patient.dmpPatientId;
+    return {
+      ...patient,
+      mrn: cmoPatientId
+        ? mrnByCmoPatientIdMap[cmoPatientId]
+        : dmpPatientId
+        ? mrnByDmpPatientIdMap[dmpPatientId]
+        : null,
+      anchorSequencingDate: dmpPatientId
+        ? anchorSeqDateByDmpPatientIdMap[dmpPatientId]
+        : null,
+    };
+  });
+}
```

Comment:
Created hashmaps of the ID mappings and anchor sequencing date data for quicker lookup when joining these PHI to the patients data.